### DISCUSSION TITLE
Apply same style rule to several columns where row expression is effectively the same

### DISCUSSION THREAD

**Discussion by arthur-shaw** [rstudio/gt#1088] [Q&A] [answered]
Ultimately, I want to flag extreme values in a several table columns. Presently, I need write code that provides a style rule for each column where extreme values are to be detected.

For example:

```
df <- tibble::tribble(
  ~a, ~b, ~c,
  1, 2, 10,
  10, 10, 1,
  5, 1, 2
)

my_style <- list(
  gt::cell_fill(color = "#F9E3D6"),
  gt::cell_text(style = "italic")
)

df |>
  gt::gt() |>
  gt::tab_style(
    style = my_style,
    locations = gt::cells_body(
      columns = a,
      rows = a >= 5
    )
  ) |>
  gt::tab_style(
    style = my_style,
    locations = gt::cells_body(
      columns = b,
      rows = b >= 5
    )
  ) |>  
  gt::tab_style(
    style = my_style,
    locations = gt::cells_body(
      columns = c,
      rows = c >= 5
    )
  ) 
```

If the expression in `rows` is different for each column, then it make sense to have one `tab_style` call per target column.

If the expression in `rows` is the same except for the variable name, I end up effectively repeating the same code one time for each target column. 

Is there a more compact way to target multiple columns with the same `rows` expression (apart from the column name)?

In effect, I would like to be able to write something like this, inspired by `dplyr::across`:

```
my_style <- list(
  gt::cell_fill(color = "#F9E3D6"),
  gt::cell_text(style = "italic")
)

df |>
  gt::gt() |>
  gt::tab_style(
    style = my_style,
    locations = gt::cells_body(
      columns = c(a, b, c),
      rows = ~ .x >= 5
    )
  )
```

**Comment by rich-iannone** [comment, marked as answer]
As of v0.8.0 of gt, we can now use `tab_style_body()` to make this a bit easier. Here's the updated code:

```r
library(gt)
library(tidyverse)

df <- tibble::tribble(
  ~a, ~b, ~c,
  1, 2, 10,
  10, 10, 1,
  5, 1, 2
)

my_style <- list(
  gt::cell_fill(color = "#F9E3D6"),
  gt::cell_text(style = "italic")
)

df |>
  gt::gt() |>
  tab_style_body(fn = function(x) x >=5, style = my_style)
```

For now, we have to use `fn = function(x) x >=5` but I'm hoping to improve it later so it accepts `fn = ~ x >=5`.

**Comment by arthur-shaw** [comment]
Many thanks, @rich-iannone ! This is a fantastic addition.

I've loved `{gt}` ever since your rstudio::conf(2019) talk, and genuinely look forward to reading each release note to see what has improved.
